tuna/rocmlir/rocmlir_tables.py

```diff
@@ -0,0 +1,269 @@
+  # pylint: disable=invalid-name ; names represent entries in job_enum column
+  # pylint: disable=duplicate-code
+  new = 1
+  running = 3
+  completed = 4
+  error = 5
+
+
+class JobMixin():
+  """Essential attributes for all jobs in the job tables"""
+
+  @declared_attr
+  def session(self):
+    """session key, as a function to connect at run time"""
+    return Column(Integer, ForeignKey("session_rocmlir.id"), nullable=False)
+
+  reason = Column(String(length=60), nullable=False, server_default="")
+  state = Column(Enum(JobEnum), nullable=False, server_default="new")
+  retries = Column(Integer, nullable=False, server_default="0")
+  result = Column(Text, nullable=True)
+
+  compile_start = Column(DateTime,
+                         nullable=False,
+                         server_default=sqla_func.now())
+  compile_end = Column(DateTime, nullable=False, server_default=sqla_func.now())
+  gpu_id = Column(Integer, nullable=False, server_default="-1")
+  machine_name = Column(String(length=60), nullable=False, server_default="")
+
+
+class ConvolutionJob(BASE, JobMixin):
+  """Represents convolutions job table"""
+  __tablename__ = "rocmlir_conv_job"
+  __table_args__ = (UniqueConstraint('config', 'session', name="uq_idx"),)
+
+  config = Column(Integer,
+                  ForeignKey("rocmlir_conv_config.id"),
+                  nullable=False,
+                  index=True)
+
+
+class ConvolutionConfig(BASE):
+  """Represents convolution config table"""
+  __tablename__ = "rocmlir_conv_config"
+
+  data_type = Column(String(length=60), nullable=False, server_default="")
+  fil_layout = Column(String(60), nullable=False, server_default="NCHW")
+  in_layout = Column(String(60), nullable=False, server_default="NCHW")
+  out_layout = Column(String(60), nullable=False, server_default="NCHW")
+  direction = Column(String(length=8), nullable=False)
+  in_channels = Column(Integer, nullable=False, server_default="0")
+  in_h = Column(Integer, nullable=False, server_default="0")
+  in_w = Column(Integer, nullable=False, server_default="0")
+  fil_h = Column(Integer, nullable=False, server_default="0")
+  fil_w = Column(Integer, nullable=False, server_default="0")
+  out_channels = Column(Integer, nullable=False, server_default="0")
+  batchsize = Column(Integer, nullable=False, server_default="0")
+  pad_h = Column(Integer, nullable=False, server_default="0")
+  pad_w = Column(Integer, nullable=False, server_default="0")
+  conv_stride_h = Column(Integer, nullable=False, server_default="0")
+  conv_stride_w = Column(Integer, nullable=False, server_default="0")
+  dilation_h = Column(Integer, nullable=False, server_default="0")
+  dilation_w = Column(Integer, nullable=False, server_default="0")
+  group_size = Column(Integer, nullable=False, server_default="0")
+  kernel_repeats = Column(Integer, nullable=False, server_default="0")
+
+  def __repr__(self) -> str:
+    return f"ConvolutionConfig {self.to_dict()}"
+
+  options = {
```

Comment:
Needs a comment.  This dict maps field names, which are also the long option names, to the short forms used in the configs.  Necessary to turn a ConvolutionConfig back into a string that we can pass to the runner.